### DISCUSSION TITLE
Enable Hardware acceleration on Linux

### DISCUSSION THREAD

**Discussion by Revival8697** [FreeTubeApp/FreeTube#4526] [General]
## Instructions to enable Hardware acceleration on Arch Linux

Also see https://github.com/FreeTubeApp/FreeTube/issues/961. To be fixed by https://github.com/FreeTubeApp/FreeTube/pull/4532.

1. [Install and verify that VA-API is working](https://wiki.archlinux.org/title/Hardware_video_acceleration).

2. Install the [nightly build](https://aur.archlinux.org/packages/freetube-nightly) of FreeTube. The nightly build has up to date Electron therefore easier to work with.

    You can use stable build but it may be harder to find the working flags.

5. Append the necessary flags:
- Copy FreeTube desktop file to user directory:

```
cp /usr/share/applications/freetube.desktop ~/.local/share/applications/freetube.desktop
```
- Check [the Arch Wiki](https://wiki.archlinux.org/index.php?title=Chromium#Hardware_video_acceleration) for the newest information on which flags are needed. 
- Test the flags by launching FreeTube with a terminal: `freetube --flag1 --flag2` and play a [test video](https://youtu.be/LXb3EKWsInQ) at 1080p60.
- For Intel GPU, use `intel_gpu_top` and see if the Video engine is being used. For others, monitor the CPU with `htop` and GPU with `nvtop`, you should see less CPU usage and more GPU usage when Hardware acceleration is working. You may have to experiment multiple times before finding the correct flags for your GPU.

    Note: As of writing, only `--enable-features=VaapiVideoDecodeLinuxGL` is needed (for my Intel 10th Gen GPU).
- Open `~/.local/share/applications/freetube.desktop` with a text editor and append the flags to the `Exec=freetube %U` line. Remember to save the file.

    For example, mine looks like this: `Exec=freetube %U --enable-features=VaapiVideoDecodeLinuxGL`.

**Comment by Revival8697** [comment]
https://github.com/FreeTubeApp/FreeTube/pull/4532 (merged) will fix this without end-user interactions.

Test builds are available at [here](https://github.com/sp1cyf0x/FreeTube/actions/runs/7474784485).

**Reply by absidue** [reply]
It's not stale, we just haven't reviewed it yet, 6 days is nothing, especially for a pull request that you'll need an actual linux machine to review it, as i doubt that trying to test hardware acceleration in a vm is going to go well.

**Comment by ngocphamm** [comment]
Just a reference point for anyone looking for it.

As of now, for version `v0.23.9 Beta` of FreeTube, under Wayland (Ubuntu 25.04), I need to add the flag `--enable-features=AcceleratedVideoDecodeLinuxGL` as in the [linked Arch Wiki](https://wiki.archlinux.org/index.php?title=Chromium#Hardware_video_acceleration) in the first post above. 

As I'm using FreeTube from Flathub, this is how I do it.

1. `cp /var/lib/flatpak/exports/share/applications/io.freetubeapp.FreeTube.desktop ~/.local/share/applications/`
2. Edit that newly copied file
3. Add `--enable-features=AcceleratedVideoDecodeLinuxGL,AcceleratedVideoDecodeLinuxZeroCopyGL` after `io.freetubeapp.FreeTube` and before `@@u`.
    * The second flag `AcceleratedVideoDecodeLinuxZeroCopyGL` doesn't seem to be needed, and/or having significant impact for me, though

Related to https://github.com/FreeTubeApp/FreeTube/pull/6274.
